jdaviz/utils.py

```diff
@@ -319,21 +336,29 @@ def flux_conversion(spec, values, original_units, target_units):
     # If there are only two values, this is likely the limits being converted, so then
     # in case we need to use the spectral density equivalency, we need to provide only
     # to spectral axis values. If there is only one value
-    if not np.isscalar(values) and len(values) == 2:
-        spectral_values = spec.spectral_axis[0]
-    else:
-        spectral_values = spec.spectral_axis
+    image_data = False
+    if spec:
+        if not np.isscalar(values) and len(values) == 2:
+            spectral_values = spec.spectral_axis[0]
+        else:
+            spectral_values = spec.spectral_axis
+
+        spec_unit = str(spec.flux.unit)
```

Comment:
I was going to say this should probably be `flux_unit`, but I guess the point is that it could be surface brightness so calling it that would also be confusing. How about `spec_y_unit`?

Reply:
For this spec unit, it's specifically looking at the Spectrum1D object. We want to see what the native units of the input data are, since that ultimately dictate the conversions/translations that take place. We can't just look at what's displayed along the spectrum viewers y-axis. I added a comment above the line that should clarify that the unit can be flux or sb, but let me know if I should add or explain this better

Reply:
Exactly. `fluxish_unit`? 😆 

I didn't realize when I wrote the original comment that this variable is only used once later in the method, so it's not a huge deal.